arelle/ValidateXbrlCalcs.py

```diff
@@ -694,55 +696,39 @@ def rangeValue(value, decimals=None, truncate=False) -> tuple[decimal.Decimal, d
-            if scale:
-                vFloat = pow(vFloat, iScale)
-    except (decimal.InvalidOperation, ValueError): # would have been a schema error reported earlier
+        valueDecimal = decimal.Decimal(value)
+    except (decimal.InvalidOperation, ValueError):  # would have been a schema error reported earlier
         return None
-    if precision is not None:
-        if not isinstance(precision, (int,float)):
-            if precision == "INF":
-                return None
-            else:
-                try:
-                    precision = int(precision)
-                except ValueError: # would be a schema error
-                    return None
-        if isinf(precision) or precision == 0 or isnan(precision) or vFloat == 0:
+    if not valueDecimal.is_normal():  # prevent exception with excessive quantization digits
+        return None
+    # Normalize decimals
+    if isinstance(decimals, str):
+        if decimals == "INF":
             return None
         else:
-            vAbs = fabs(vFloat)
-            log = log10(vAbs)
-            decimals = precision - int(log) - (1 if vAbs >= 1 else 0)
-    elif decimals is not None:
-        if not isinstance(decimals, (int,float)):
-            if decimals == "INF":
+            try:
+                decimals = int(decimals)
+            except ValueError:  # would have been a schema error reported earlier
                 return None
-            else:
-                try:
-                    decimals = int(decimals)
-                except ValueError: # would be a schema error
-                    return None
-        if isinf(decimals) or isnan(decimals):
-            return None
-    else:
+    if isinf(decimals) or isnan(decimals):
+        return None
+    if decimals > 0:
+        divisor = ONE.scaleb(-decimals)  # fractional scaling doesn't produce scientific notation
+    else:  # extra quantize step to prevent scientific notation for decimal number
+        divisor = ONE.scaleb(-decimals).quantize(ONE, decimal.ROUND_HALF_UP) # should never round
+    try:
+        quotient, insignificant = divmod(valueDecimal, divisor)
+    except decimal.InvalidOperation:
         return None
-    if vDecimal.is_normal() and -28 <= decimals <= 28: # prevent exception with excessive quantization digits
-        if decimals > 0:
-            divisor = ONE.scaleb(-decimals) # fractional scaling doesn't produce scientific notation
-        else:  # extra quantize step to prevent scientific notation for decimal number
-            divisor = ONE.scaleb(-decimals).quantize(ONE, decimal.ROUND_HALF_UP) # should never round
-        insignificantDigits = abs(vDecimal) % divisor
-        if insignificantDigits:
-            return (vDecimal // divisor * divisor,  # truncated portion of number
-                    insignificantDigits)   # nsignificant digits portion of number
+    if insignificant:
+        significant = quotient * divisor
+        return significant, insignificant
```

Comment:
Do we need to `abs` the divisor for negative values?

Reply:
To match previous behavior, yes. The values are only output in error messages as far as I can tell but negative insignificant digit values do seem strange. I've also added negative value test cases.